crates/aranya-policy-lang/src/lang/parse/error.rs

```diff
@@ -50,28 +50,39 @@ pub enum ParseErrorKind {
 #[derive(Debug, Clone)]
 pub struct ParseError {
     pub kind: ParseErrorKind,
-    message: String,
+    pub message: String,
+    /// Line and column location of the error, if available.
+    pub location: Option<(usize, usize)>,
 }
 
 impl ParseError {
     pub(crate) fn new(kind: ParseErrorKind, message: String, span: Option<Span<'_>>) -> ParseError {
-        let prefix = match span {
-            Some(s) => {
-                let text = s.as_str();
-                let (line, col) = s.start_pos().line_col();
-                format!("line {line} column {col}: {text}: ")
-            }
-            None => String::from(""),
-        };
+        let location = span.map(|s| s.start_pos().line_col());
         ParseError {
             kind,
-            message: format!("{prefix}{message}"),
+            message,
+            location,
+        }
+    }
+
+    /// Return a new error with a location starting from the given line.
+    pub fn adjust_line_number(&self, start_line: usize) -> ParseError {
+        ParseError {
+            kind: self.kind.clone(),
+            message: self.message.clone(),
+            location: self
+                .location
+                .map(|(line, col)| (line.saturating_add(start_line), col)),
```

Comment:
Using `saturating_add` feels a bit sketchy, but the effort to handle the `Option<Result<>>` doesn't seem worth it here. How likely are we to ever overflow a `usize` number of lines?